File: PythonParticipant/registrationclient/RegistrationClient.py
from thriftgen.registration.RegistrationService import *
from thriftlib.Thrift import TException
from thriftlib.transport import *
from thriftlib.protocol import TBinaryProtocol


class RegistrationClient:
    def __init__(self, serverUrl, port):
        self.registrationServerUrl = serverUrl
        self.registrationPort = port


    def register(self, coordinationContext, protocolIdentifier, participantEPR):
        try:
            transport = TSocket.TSocket(self.registrationServerUrl, self.registrationPort)
            protocol = TBinaryProtocol.TBinaryProtocol(transport)
            registrationClient = Client(protocol)
            print "register called."

            transport.open()
            print "open "
            result = registrationClient.registerParticipant(coordinationContext, protocolIdentifier, participantEPR);

            transport.close()
            print ("registration finished...")

        except TException as tx:
            print ("%s" % tx.message)

        return result

File: Wso2Greg/org.wso2.carbon.registry.acidothinterface/org.wso2.carbon.registry.acidothinterface_2/src/main/java/org/wso2/carbon/registry/acidothinterface/registrationclient/RegistrationClient.java
package org.wso2.carbon.registry.acidothinterface.registrationclient;

import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;
import org.wso2.carbon.registry.acidothinterface.thriftgen.registration.CoordinationContext;
import org.wso2.carbon.registry.acidothinterface.thriftgen.registration.EndPointReference;
import org.wso2.carbon.registry.acidothinterface.thriftgen.registration.RegistrationService;


public class RegistrationClient {
	
	private String serverUrl;
    private int port;
	
	public RegistrationClient(String serverUrl, int port){
		this.serverUrl = serverUrl;
        this.port = port;
	}
	
	public boolean Register(CoordinationContext coorcontext, String protocolIdentifier, EndPointReference protocolServiceEPR){
		
		TTransport transport;
		RegistrationService.Client registrationClient;
		boolean result = false;
		
		transport = new TSocket(serverUrl, port);
		try {
			
			transport.open();
			TProtocol protocol = new TBinaryProtocol(transport);
			registrationClient = new RegistrationService.Client(protocol);
			
			result = registrationClient.registerParticipant(coorcontext, protocolIdentifier, protocolServiceEPR);
			
			transport.close();
			
			}catch (TException e) {
				e.printStackTrace();			
		}
		
		return result;
	}

}
